kadai-adapter/src/main/java/io/kadai/adapter/monitoring/HealthCheck.java

```diff
@@ -0,0 +1,199 @@
+      outboxHealth
+          .withDetail("Outbox Service", outboxResponse.getBody())
+          .status(Health.up().build().getStatus());
+    } catch (Exception e) {
+      outboxHealth
+          .withDetail("Outbox Service Error", e.getMessage())
+          .status(Health.down().build().getStatus());
+    }
+
+    try {
+      ResponseEntity<VersionInfoRepresentationModel> kadaiResponse =
+          pingKadaiGetCurrentSchemaVersion();
+      kadaiHealth
+          .withDetail("Kadai Service", kadaiResponse.getBody())
+          .status(Health.up().build().getStatus());
+    } catch (Exception e) {
+      kadaiHealth
+          .withDetail("Kadai Service Error", e.getMessage())
+          .status(Health.down().build().getStatus());
+    }
+
+    return Health.status("External Services Status")
+        .withDetail("Camunda Health", camundaHealth.build())
+        .withDetail("Outbox Health", outboxHealth.build())
+        .withDetail("Kadai Health", kadaiHealth.build())
+        .build();
+  }
+
+  private ResponseEntity<EngineInfoRepresentationModel[]> pingCamundaRest() throws Exception {
+    String address = env.getProperty("camundaOutboxService.address");
+    String port = env.getProperty("camundaOutboxService.port");
+
+    String url = address + ":" + port + "/example-context-root/engine-rest/engine";
+    try {
+      ResponseEntity<EngineInfoRepresentationModel[]> response =
+          restTemplate.exchange(
+              url,
+              HttpMethod.GET,
+              null,
+              ParameterizedTypeReference.forType(EngineInfoRepresentationModel[].class));
+      return response;
+    } catch (Exception e) {
+      throw new Exception("Error connecting to Camunda REST API: " + e.getMessage(), e);
+    }
+  }
+
+  private ResponseEntity<EventCountRepresentationModel> pingOutBoxRestGetNumberFailedEvents()
+      throws Exception {
+    String address = env.getProperty("camundaOutboxService.address");
+    String port = env.getProperty("camundaOutboxService.port");
+
+    String url = address + ":" + port + "/example-context-root/outbox-rest/events/count?retries=0";
+    try {
+
+      ResponseEntity<EventCountRepresentationModel> response =
+          restTemplate.exchange(
+              url,
+              HttpMethod.GET,
+              null,
+              ParameterizedTypeReference.forType(EventCountRepresentationModel.class));
+      return response;
+    } catch (Exception e) {
+      throw new Exception(
+          "Error getting number of failed events from Outbox: " + e.getMessage(), e);
+    }
+  }
+
+  private ResponseEntity<VersionInfoRepresentationModel> pingKadaiGetCurrentSchemaVersion()
+      throws Exception {
+    String address = env.getProperty("kadaiService.address");
+    String port = env.getProperty("kadaiService.port");
+
+    String url = address + ":" + port + "/kadai/api/v1/version";
+
+    HttpEntity<?> auth = new HttpEntity<>(generateHeadersForUser("admin"));
```

Comment:
How is the method `pingKadaiGetCurrentSchemaVersion` supposed to work, if there isn't an "admin" user, or if the "admin" user has changed password, as every "admin" user in the world should do?

Reply:
But if we want to call the current Schema Version of Kadai, we have to have admin permission